packages/nimble-components/src/table/tests/table-selection.spec.ts

```diff
@@ -0,0 +1,409 @@
+import { html } from '@microsoft/fast-element';
+import type { Table } from '..';
+import { waitForUpdatesAsync } from '../../testing/async-helpers';
+import { createEventListener } from '../../utilities/tests/component';
+import { type Fixture, fixture } from '../../utilities/tests/fixture';
+import {
+    TableRecord,
+    TableRowSelectionEventDetail,
+    TableRowSelectionMode
+} from '../types';
+import { TablePageObject } from './table.pageobject';
+
+interface SimpleTableRecord extends TableRecord {
+    stringData: string;
+}
+
+const simpleTableData = [
+    {
+        stringData: '0',
+        stringData2: 'a'
+    },
+    {
+        stringData: '1',
+        stringData2: 'b'
+    },
+    {
+        stringData: '2',
+        stringData2: 'c'
+    }
+] as const;
+
+// prettier-ignore
+async function setup(): Promise<Fixture<Table<SimpleTableRecord>>> {
+    return fixture<Table<SimpleTableRecord>>(
+        html`<nimble-table>
+            <nimble-table-column-text field-name="stringData">stringData</nimble-table-column-text>
+        </nimble-table>`
+    );
+}
+
```

Comment:
Can we have some tests going through the pre-connect, imperative workflow:

```ts
table = createElement('nimble-table')
column = createElement('column')
table.append(column)
await setData()
// different selection tests
```

Ran into this neat pattern for setting js properties during imperative node creation which can help reduce verbosity a bit:

<img width="314" alt="image" src="https://user-images.githubusercontent.com/1588923/230257179-a31d686a-8389-4536-a87d-8c8b434cb422.png">